docs/using/recipes.md

```diff
@@ -123,17 +96,13 @@ RUN mamba create --quiet --yes -p "${CONDA_DIR}/envs/${conda_env}" python=${py_v
 #     mamba env create -p "${CONDA_DIR}/envs/${conda_env}" -f environment.yml && \
 #     mamba clean --all -f -y
 
-
-# create Python 3.x environment and link it to jupyter
+# create Python kernel and link it to jupyter
 RUN "${CONDA_DIR}/envs/${conda_env}/bin/python" -m ipykernel install --user --name="${conda_env}" && \
     fix-permissions "${CONDA_DIR}" && \
     fix-permissions "/home/${NB_USER}"
 
 # any additional pip installs can be added by uncommenting the following line
-# RUN "${CONDA_DIR}/envs/${conda_env}/bin/pip" install
-
-# prepend conda environment to path
-ENV PATH "${CONDA_DIR}/envs/${conda_env}/bin:${PATH}"
```

Comment:
Don't figure why it is necessary to prepend the env `PATH`. `PATH` manipulation is handled by the `conda activate`.

Reply:
I might be wrong here, but what happens, if someone creates a new py3 environment with his favourite python version and then runs `python3` command .

I think, if we prepend to PATH, then the favourite python3 will be run.
If we append to `.bashrc`, does it work for the next `RUN` step?
Also, does it work, when we change the `$NB_USER` and actually run the docker using `--user root`?


That's why I'm quite sure prepending to PATH does what expected and I'm not confident about .bashrc.
Could you check this out, please?

Reply:
Hello,

Here my answers.

> I might be wrong here, but what happens, if someone creates a new py3 environment with his favourite python version and then runs `python3` command .

It will run the `python3` executable corresponding to the activated conda environment.

> If we append to .bashrc, does it work for the next RUN step?

No but there is no additional `RUN` step and the previous uses a full path call `"${CONDA_DIR}/envs/${conda_env}/bin/pip"`. So I do not think it's a good reason to keep it.

> Also, does it work, when we change the `$NB_USER` and actually run the docker using `--user root`?
> [...] Could you check this out, please?

Sure, it works as expected with this command.

```
docker run -it --rm \     
    -p 8888:8888 \
    --user root \
    -e NB_USER="romain" \
    -e CHOWN_HOME=yes \
    -w "/home/${NB_USER}" jupyter/custom-notebook
```

IMO you have two option tuning the `PATH` yourself or let `conda` activation script do it for you and also take care to perform other activation actions (source additional variables for example). I do not think it's a good idea to mix these two modes.

> Activating environments is essential to making the software in the environments work well. Activation entails two primary functions: adding entries to PATH for the environment and running any activation scripts that the environment may contain .

You can have a look to [the `conda active` doc](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#activating-an-environment) to make you own opinion.

Best

Reply:
Why are you requiring a change before asking the question?

Reply:
Sorry, I should have pressed comment.
Probably should go to sleep :)
Sorry once again ❤️

Reply:
Not a problem but I'm investing time here to help and especially on this follow-on PR you asked me to do. I try always to be nice but it's a bit harsh to receive this kind of review.

Reply:
@mathbunnyru it's Ok and my English is not my greatest strength too (I hope 😄). We have worked together here for a long time now and your investment is impressive. So I hope we will continue to work in a good mood like we do in the past. I think we have sometimes different approaches, a complementary knowledges, but not the same investment now. However, we should try to keep this good collaboration. On my side I will try to do it even if I invest less time here than you do. Working here for free should continue to be pleasant and even fun while doing things right like we always try to do. So I'm going to bed too 💤, have a nice sleep 😄.